main.tf

```diff
@@ -8,7 +8,7 @@ resource "aws_kms_key" "backup" {
 }
 
 resource "aws_kms_alias" "backup" {
-  name          = "alias/aws_backup-${var.vault_name}"
+  name          = var.kms_key_alias == null ? var.kms_key_alias : "alias/aws_backup-${var.vault_name}-${var.env}"
```

Comment:
maybe you need the following reverse check here, to use your custom passed kms key name when it is not null?
```terraform
name = var.kms_key_alias != null ? var.kms_key_alias : "alias/aws_backup-${var.vault_name}-${var.env}"
```